File: python/Mahasiswa.py
from CivitasAkademik import CivitasAkademik
from ProgramStudi import ProgramStudi

class Mahasiswa(CivitasAkademik):
# kelas yang digunakan untuk mengimplemntasikan kelas Mahasiswa
    
    # inisialisasi atribut
    __NIM = 0
    __fakultas = ""
    __prodi = ProgramStudi("","","") # has a

    def __init__(self, nik=0, nama="", jenis_kelamin="", asal_univ="", email="", nim=0, fakultas="", prodi=""):
    # konstruktor
        super().__init__(nik, nama, jenis_kelamin, asal_univ, email)
        self.__NIM = nim
        self.__fakultas = fakultas
        self.__prodi = prodi

    # setter & getter methods

    def setNIM(self, nim):
    # mengset nilai atribut NIM
        self.__NIM = nim

    def setFakultas(self, fakultas):
    # mengset nilai atribut fakultas
        self.__fakultas = fakultas
        
    def setProdi(self, prodi):
        self.__prodi = prodi
    
    def getNIM(self):
    # mengmbalikan nilai atribut NIM
        return self.__NIM

    def getFakultas(self):
    # mengmbalikan nilai atribut fakultas
        return self.__fakultas

    def getProdi(self):
    # mengmbalikan nilai atribut prodi
        return self.__prodi

File: c++/Mahasiswa.cpp
#include "CivitasAkademik.cpp"
#include "ProgramStudi.cpp"

class Mahasiswa : public CivitasAkademik {
    private:
        int NIM;
        string fakultas;
        ProgramStudi prodi;

    public:
        Mahasiswa() {}

        Mahasiswa(int NIK, string nama, string jenis_kelamin, string asal_universitas, string email_edu, int NIM, string fakultas, ProgramStudi prodi) : CivitasAkademik(NIK, nama, jenis_kelamin, asal_universitas, email_edu) {
            this->NIM = NIM;
            this->fakultas = fakultas;
            this->prodi = prodi;
        }

        /* setter */
        void setNIM(int nim) {
            this->NIM = nim;
        }

        void setFakultas(string fakultas) {
            this->fakultas = fakultas;
        }

        void setProdi(ProgramStudi prodi) {
            this->prodi = prodi;
        }

        /* getter */
        int getNIM() {
            return NIM;
        }

        string getFakultas() {
            return fakultas;
        }

        ProgramStudi getProdi() {
            return prodi;
        }

        ~Mahasiswa() {}
};